src/plugins/diagram-viewer/diagram-toolbar-buttons.tsx

```diff
@@ -10,6 +10,9 @@ import { TileModelContext } from "../../components/tiles/tile-api";
 import { DiagramTileMethodsContext } from "./diagram-tile";
 import { useUIStore } from "../../hooks/use-stores";
 import { kNewVariableButtonDraggableId } from "./diagram-types";
+import { kGraphTileType } from "../graph/graph-defs";
```

Comment:
Will this always bring in the graph when the diagram tile is included?

Reply:
If graph-defs has no imports then it shouldn't cause a problem. However we have planned to add a lint rule that wouldn't allow one plugin to import files from another plugin to enforce the isolation. 

One possibility to avoid the dependency (but not help type safety), is to change the button to be a `ViewVariablesButton`. And then in the configuration of the button the graph type is passed. So then if we add another tile which can view variables this same button could be used.

I had intended for the view data button for tables and datacards to work this way, and it would even pick up the icon from the tile type and then overlay it with the 'eye' badge. I don't know if that implementation stuck around though.